clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -1392,4 +1356,16 @@ public void registerStateListener(MemberStateListener listener) {
         }
         this.stateUpdatesListeners.add(listener);
     }
+
+    /**
+     * If either a new target assignment or new metadata is available that we have not yet attempted
+     * to reconcile, and we are currently in state RECONCILING, trigger reconciliation.
+     */
+    @Override
+    public PollResult poll(final long currentTimeMs) {
+        if (state == MemberState.RECONCILING) {
+            maybeReconcile();
```

Comment:
I have one remaining question for my understanding. If there is unresolved topic ids, is it going to request a metadata update (in `findResolvableAssignmentAndTriggerMetadataUpdate`) every time that `poll` is called? I believe that it won't be the case because the `Metadata` class will request a full update only once. Is my understanding correct?

Reply:
Yes, this is also my understanding.

Reply:
Just for the record you are both right, us requesting metadata updates from the reconciliation, is only setting a flag in the metadata object indicating that it needs a full update, but effectively a single metadata request is going to be issued when that flag is on, on the next network client poll. Here btw, we just have a related jira for a potential future improvement, regarding partial metadata updates when reconciling (https://issues.apache.org/jira/browse/KAFKA-15847), since right now we request it for all topics, even if we only need the ones involved in the assignment.